Backend/Hub/UserHub.cs

```diff
@@ -53,9 +51,46 @@ public async Task<string> GetPrevChallengeName()
             return user.PrevChallenges.LastOrDefault().ToString();
         }
       
-        public Task SendMetric(FrontendMetric metric, string data)
+        public async Task SendMetric(FrontendMetric metric, string data)
         {
-            throw new NotImplementedException();
+
+            UserSession session = await Database.UserSessions.FromSignalRId(Context.UserIdentifier);
+            User user = session.User;
+
+            switch (metric)
```

Comment:
Maybe instead of this switch, consider making a Dictionary<FrontendMetric, SessionLogEntryType> so it runs in constant time and there is only one method call (cleaner code)